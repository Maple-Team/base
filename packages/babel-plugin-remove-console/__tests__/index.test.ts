/* file: test/unit.test.js */

import pluginTester from 'babel-plugin-tester'
import removeConsolePlugin from '../src/'

pluginTester({
  plugin: removeConsolePlugin,
  pluginOptions: { exclude: ['debug', 'error', 'warn'] },
  pluginName: '@liutsing/babel-plugin-remove-console',
  snapshot: true,
  presets: ['@babel/preset-env', '@babel/preset-typescript'],
  babelOptions: {
    filename: __filename,
  },
  tests: [
    {
      code: 'console.log(1)',
      output: '',
    },
    {
      snapshot: false,
      code: 'console.debug(1)',
      output: 'console.debug(1)',
    },
    {
      snapshot: false,
      code: 'console.warn(1)',
      output: 'console.warn(1)',
    },
    {
      snapshot: false,
      code: 'console.error(1)',
      output: 'console.error(1)',
    },
  ],
})
