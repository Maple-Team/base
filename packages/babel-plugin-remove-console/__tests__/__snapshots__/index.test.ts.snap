// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@liutsing/babel-plugin-remove-console 1. @liutsing/babel-plugin-remove-console: 1. @liutsing/babel-plugin-remove-console 1`] = `

console.log(1)

      ↓ ↓ ↓ ↓ ↓ ↓



`;
